### PR TITLE
Upgrade to Go 1.19

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

* `crd:trivialVersions` is deprecated in controller-tool v0.7.0. See https://github.com/kubernetes-sigs/controller-tools/pull/607/ for more details.

* `go.mod` / `go.sum`: Update `golang:1.17` to `golang:1.19` in `go.mod`, and then run the `go mod tidy` command to add missing modules and remove unused ones.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: scripts/install-docker.sh [hunk 1]
```diff
@@ -4,7 +4,4 @@
 export DEBIAN_FRONTEND=noninteractive
 export TZ=America/Los_Angeles
 
-apt-get update -qq && apt-get upgrade -qq
```
Comment (kevin85421) [inline scripts/install-docker.sh:7]: In `.buildkite/test-rayjob-sample-yamls.yml`, a `curl` command is executed before `install-docker.sh`, so there's no need to install curl again.

File: .buildkite/test-rayjob-sample-yamls.yml [hunk 2]
```diff
@@ -60,7 +60,9 @@
 
     # Install python 3.10 and pip
     - apt-get update
-    - apt-get install -y python3.10 python3-pip
```
Comment (kevin85421) [inline .buildkite/test-rayjob-sample-yamls.yml:63]: `apt-get install -y python3-pip` will report ["error: externally-managed-environment"](https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-everytime-i-use-pip3) after I upgrade the image to `golang:1.19`.

**Comment by kevin85421** [comment]
cc @anishasthana @blublinsky would you mind taking a look? Thx!

**Review by architkulkarni — APPROVED** [review]

**Review by blublinsky — APPROVED** [review]
LGTM

File: apiserver/go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/ray-project/kuberay/apiserver
 
-go 1.17
+go 1.19
```
Comment (tedhtchang) [inline apiserver/go.mod:3]: @kevin85421 may need to update the go version in the table https://github.com/ray-project/kuberay/blob/master/apiserver/DEVELOPMENT.md#kuberay-api-server-user-guide

**Pull request merged**